### PR TITLE
Basic PHP/Apache/SQL with composer

### PR DESCRIPTION
**Pull request by mibst**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,16 @@
+FROM php:7.2-apache
+COPY app/ /var/www/html/
+COPY composer.json /var/www/html/
+
+WORKDIR /root
```
Comment (javier-lopez) [inline Dockerfile]: Why is this necesary?

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+RUN apt-get update -y && apt-get upgrade -y && apt-get install git -y &&\
+    curl https://raw.githubusercontent.com/composer/getcomposer.org/11a3bc677c1e6f2b120578d61b780aed7bfdff58/web/installer | php -- --quiet &&\
```
Comment (javier-lopez) [inline Dockerfile]: is there any way to download latest?, or even better download a specific version?, maybe we could use a system variable?, this seems to be hard coded to a specific composer release

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+    mv composer.phar /usr/local/bin/composer &&\
+    chmod +x /usr/local/bin/composer
+
+WORKDIR /var/www/html
+
+RUN composer update
```
Comment (javier-lopez) [inline Dockerfile:11]: Would prefer to set a specific composer version, see comment above

File: docker-compose.yml [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+  mysql:
+    image: mysql:5.7
+    environment:
+      - MYSQL_DATABASE=${DB_NAME:-test_db}
+      - MYSQL_USER=${DB_USER:-test_user}
+      - MYSQL_PASSWORD=${DB_PASS:-test_pass}
+      - MYSQL_ROOT_PASSWORD=${DB_ROOT_PASS:-super_secret_pass}
+    ports:
+      - "9906:3306"
```
Comment (javier-lopez) [inline docker-compose.yml]: Why this port binding is different but postgresql uses the same?, "9906:3306" vs "5433:5432"

File: docker-compose.yml [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+    profiles:
+      - mysql
+
+  phpmyadmin:
+    image: phpmyadmin
```
Comment (javier-lopez) [inline docker-compose.yml]: could you provide a specific image version?

File: docker-compose.yml [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+      - POSTGRES_USER=${DB_USER:-test_user}
+      - POSTGRES_PASSWORD=${DB_PASS:-test_pass}
+    ports:
+      - "5433:5432"
+    profiles:
+      - postgres
+      - postgresql
+
+  pgadmin:
+    image: 'thajeztah/pgadmin4'
```
Comment (javier-lopez) [inline docker-compose.yml:52]: could you provide a specific image version?, also, what advantages has this image over the official?, https://hub.docker.com/r/dpage/pgadmin4/

File: setup.sh [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+        DB_HOST="postgres:5432"
+        ;;
+    *)
+        >&2 echo "Choose one of: mysql, postgres"
+        exit 1
+        ;;
+esac
+
+if [ $# -gt 2 ] && [ -f ${2} ]; then
+    set -o allexport
```
Comment (javier-lopez) [inline setup.sh:25]: what is this for?, could you add a #comment describing why it's required?

**Review by javier-lopez — CHANGES_REQUESTED** [review]
Great work!, left some comments.

File: Dockerfile (hunk 1 above)
Comment (javier-lopez) [reply Dockerfile]: Refering to the `WORKDIR /root` part

File: Dockerfile (hunk 3 above)
Comment (javier-lopez) [reply Dockerfile:11]: refering to `RUN composer update`

File: Dockerfile (hunk 1 above)
Comment (mibst) [reply Dockerfile]: I used it when the next commands were downloading checksums and storing them on the filesystem. Should be removed.

File: Dockerfile (hunk 2 above)
Comment (mibst) [reply Dockerfile]: It downloads latest, following the [docs](https://getcomposer.org/doc/faqs/how-to-install-composer-programmatically.md). Not sure how could we download a specific version.

File: docker-compose.yml (hunk 4 above)
Comment (mibst) [reply docker-compose.yml]: In the test machine the mysql port was in use, forgot to forward it to the correct one on push. Will be corrected.

File: docker-compose.yml (hunk 5 above)
Comment (mibst) [reply docker-compose.yml]: Will be corrected.
```suggestion
    image: phpmyadmin:5.1.1-apache
```

File: docker-compose.yml (hunk 6 above)
Comment (mibst) [reply docker-compose.yml:52]: Ease of config, but will use the official once I figure out how to configure it correctly (docs are not great)

File: setup.sh (hunk 7 above)
Comment (mibst) [reply setup.sh:25]: It's being use as a shorthand to export the variables defined in the file passed as a .env file.
```suggestion
    # export all variables in env file
    set -o allexport
```

File: Dockerfile (hunk 2 above)
Comment (javier-lopez) [reply Dockerfile]: Ok, I think a better alternative would be to get composer from the official docker compose container: https://hub.docker.com/_/composer?tab=tags&page=1&ordering=last_updated

`COPY --from=composer:latest /usr/bin/composer /usr/bin/composer`

latest could be replaced with a version variable

File: docker-compose.yml (hunk 6 above)
Comment (javier-lopez) [reply docker-compose.yml:52]: It's ok, easy of usage can be a good reason, do not spend a lot of time trying to figure out how to use the official image if it's that hard to configure, just be careful with security, it's ok in development mode, not so great if we push eventually images based on this configuration to prod

**Review by javier-lopez — APPROVED** [review]
LGTM, :shipit:

**Pull request merged**